numpy/core/fromnumeric.py

```diff
@@ -1795,6 +1795,11 @@ def nonzero(a):
         Equivalent ndarray method.
     count_nonzero :
         Counts the number of non-zero elements in the input array.
+        
+    Notes
+    -----
+    To obtain the non-zero values, it is recommended to use ``x[y.astype(bool)]`` 
+    which will correctly handle 0-d arrays.    
```

Comment:
The description you use here makes x vs y look like a typo. It would be good to include the form you're recommending against here too.

Also, worth commenting that in most cases y is already a boolean array, so as type is unnecessary

Reply:
@abhinavsagar could you follow up on this, and the other comments? I think it would be good to keep the old comment about the values and move it here, as well as mentioning the `astype` is not necessary.

The story would be that you can get the values using the old statement (with the exception of 0d), but if you only need the values, you should prefer the other statement.

Reply:
@seberg Done the required changes.